shared/src/commonMain/kotlin/io/customer/shared/di/KMMComponent.kt

```diff
@@ -39,6 +40,19 @@ class KMMComponent(
         staticComponent.attachSDKConfig(config = sdkComponent.customerIOConfig)
     }
 
+    val backgroundQueue: BackgroundQueue
+        get() = getSingletonInstance {
```

Comment:
I made [a comment](https://github.com/customerio/mobile-shared/pull/4#discussion_r1052480955) in another KMM PR about defaulting to non-singletons. Wanted to point that out in this PR, too so depending on what we decide in that other PR, changes do not get unnoticed in this PR. 

So, no action here until [this discussion](https://github.com/customerio/mobile-shared/pull/4#discussion_r1052480955) gets resolved. 

This comment applies to all other places in this class that use singletons.

Reply:
I don't think we will need multiple queues, we currently have this [singleton in Android SDK](https://github.com/customerio/customerio-android/blob/6da8b8d3d16fa1c9c0acdb6012271a2252b30951/sdk/src/main/java/io/customer/sdk/di/CustomerIOComponent.kt#L60-L72) too. Irrespective of what we decide on the linked discussion, the queue will remain singleton as it would become very challenging to manage multiple queues.